src/dataset.cpp

```diff
@@ -504,10 +532,10 @@ Dataset &Dataset::operator-=(const double value) {
 }
 
 Dataset &Dataset::operator*=(const Dataset &other) {
-  return times_equals(*this, other);
+  return op_equals(*this, other, &aligned::multiply, &aligned::multiply);
```

Comment:
@igudich mentioned inconsistency with c-style function pointers (as per PR description), I'll open a new issue around this to move to a templated functor or similar